motr/ut/idx_dix.c

```diff
@@ -781,15 +781,262 @@ struct m0_client* st_get_instance()
+		srv_srv = m0_reqh_service_lookup(srv_reqh, &srv_dtm0_fid);
+		rc = m0_dtm0_service_process_connect(srv_srv, &cli_srv_fid,
+						     cl_ep_addr, false);
+		M0_UT_ASSERT(rc == 0);
+	}
+
+	general_ifid_fill(&duc.duc_ifid, true);
+
+	/* Save the context */
+	duc.duc_cli_svc_fid = cli_srv_fid;
+	duc.duc_srv_svc_fid = srv_dtm0_fid;
+	duc.duc_srv_svc = srv_srv;
+
+	return 0;
+}
+
+static void idx_setup(void)
+{
+	struct m0_op            *op = NULL;
+	struct m0_idx           *idx = &duc.duc_idx;
+	struct m0_container     *realm = &duc.duc_realm;
+	int                      rc;
+	struct m0_fid           *ifid = &duc.duc_ifid;
+
+	m0_idx_init(idx, &realm->co_realm, (struct m0_uint128 *) ifid);
+
+	/* Create the index */
+	rc = m0_entity_create(NULL, &idx->in_entity, &op);
+	M0_UT_ASSERT(rc == 0);
+	m0_op_launch(&op, 1);
+	rc = m0_op_wait(op, M0_BITS(M0_OS_STABLE), WAIT_TIMEOUT);
+	M0_UT_ASSERT(rc == 0);
+	m0_op_fini(op);
+	m0_op_free(op);
+	op = NULL;
+}
+
+static void idx_teardown(void)
+{
+	struct m0_op            *op = NULL;
+	int                      rc;
+
+	/* Delete the index */
+	rc = m0_entity_delete(&duc.duc_idx.in_entity, &op);
+	M0_UT_ASSERT(rc == 0);
+	m0_op_launch(&op, 1);
+	rc = m0_op_wait(op, M0_BITS(M0_OS_STABLE), WAIT_TIMEOUT);
+	M0_UT_ASSERT(rc == 0);
+	m0_op_fini(op);
+	m0_op_free(op);
+	m0_idx_fini(&duc.duc_idx);
+	M0_SET0(&duc.duc_idx);
+}
+
+
+static int duc_teardown(void)
+{
+	struct m0_reqh_service  *srv_srv = duc.duc_srv_svc;
+	struct m0_fid            cli_srv_fid  = duc.duc_cli_svc_fid;
+	int                      rc;
+
+	if (ENABLE_DTM0) {
+		/* Disconnect from the server */
+		rc = m0_dtm0_service_process_disconnect(srv_srv, &cli_srv_fid);
+		M0_UT_ASSERT(rc == 0);
+	}
+
+	rc = ut_suite_mt_idx_dix_fini();
+	m0_fi_disable("m0_dtm0_in_ut", "ut");
+	return rc;
+}
+
+/* Submits multiple M0 client (PUT|DEL) operations and then waits on "phase1"
+ * states, and then waits on "phase2" states.
+ */
+static void run_m0ops(uint64_t nr, enum m0_idx_opcode opcode,
+		      uint64_t phase1wait,
+		      uint64_t phase2wait)
+{
+	struct m0_idx      *idx = &duc.duc_idx;
+	struct m0_op      **ops;
+	int                *rcs;
+	struct m0_bufvec   *key_vecs;
+	char               *val = NULL;
+	struct m0_bufvec    vals = {};
+	m0_bcount_t         len = 1;
+	int                 flags = 0;
+	uint64_t            i;
+	int                 rc;
+
+	M0_PRE(M0_IN(opcode, (M0_IC_PUT, M0_IC_DEL)));
+	M0_ALLOC_ARR(ops, nr);
+	M0_UT_ASSERT(ops != NULL);
+	M0_ALLOC_ARR(rcs, nr);
+	M0_UT_ASSERT(rcs != NULL);
+	M0_ALLOC_ARR(key_vecs, nr);
+	M0_UT_ASSERT(key_vecs != NULL);
+
+	if (opcode == M0_IC_PUT) {
+		val = m0_strdup("ItIsAValue");
+		M0_UT_ASSERT(val != NULL);
+		vals = M0_BUFVEC_INIT_BUF((void **) &val, &len);
+	}
+
+	/* Execute the ops */
+	for (i = 0; i < nr; ++i) {
+		rc = m0_bufvec_alloc(&key_vecs[i], 1, sizeof(i));
+		M0_UT_ASSERT(key_vecs[i].ov_vec.v_count[0] == sizeof(i));
+		memcpy(key_vecs[i].ov_buf[0], &i, sizeof(i));
+
```

Comment:
[suggestion] Can we add an assert for rc to make codacy successful.
 M0_UT_ASSERT(rc == 0);

Reply:
Thanks. Will be fixed in a separate PR, as it was agreed.